dynamiqs/integrators/sepropagator/expm_integrator.py

```diff
@@ -26,46 +25,33 @@ def __init__(self, *args):
             )
 
     def run(self) -> PyTree:
-        # === find times at which the Hamiltonian changes (between t0 and tfinal)
-        t0 = jnp.asarray(self.t0).reshape(-1)
-        times = _concatenate_sort(t0, self.H.discontinuity_ts, self.ts)
-        inbounds = (times[:-1] >= t0) & (times[1:] <= self.ts[-1])
-        t_diffs = jnp.where(inbounds, jnp.diff(times), 0.0)
+        # === find all times at which to stop in [t0, t1]
+        # find all times where the solution should be saved (self.ts) or at which the
+        # Hamiltonian changes (self.H.discontinuity_ts)
+        disc_ts = self.H.discontinuity_ts
+        if disc_ts is not None:
+            disc_ts = disc_ts.clip(self.t0, self.t1)
+        times = _concatenate_sort(jnp.asarray([self.t0]), self.ts, disc_ts)  # (ntimes,)
```

Comment:
What is the reason for `jnp.asarray([self.t0])` as opposed to `jnp.asarray(self.t0).reshape(-1)`? I used the latter because of the comment here https://github.com/dynamiqs/dynamiqs/blob/main/dynamiqs/integrators/core/propagator_integrator.py#L50 .

Reply:
Both options work in this context , I chose `jnp.asarray([self.t0])` so that it fits on a single line 😇 Ideally we should convert `t0` upfront in `dq.Options` so we don't have to worry about this downstream.